### ISSUE TITLE
java.lang.ClassNotFoundException

### ISSUE THREAD

**Issue by ukhangmu** [apereo/java-cas-client#296]
Hello,

After installing and configuring CAS for Atlassian Confluence, I am getting the following error:

```
2019-05-06 15:54:57,227 ERROR [Catalina-utility-1] [ContainerBase.[Standalone].[localhost].[/]] log Error configuring application listener of class [org.jasig.cas.client.session.SingleSignOutHttpSessionListener]
java.lang.ClassNotFoundException: org.jasig.cas.client.session.SingleSignOutHttpSessionListener
	at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1363)
	at org.apache.catalina.loader.WebappClassLoaderBase.loadClass(WebappClassLoaderBase.java:1186)
	at org.apache.catalina.core.DefaultInstanceManager.loadClass(DefaultInstanceManager.java:540)
	at org.apache.catalina.core.DefaultInstanceManager.loadClassMaybePrivileged(DefaultInstanceManager.java:521)
	at org.apache.catalina.core.DefaultInstanceManager.newInstance(DefaultInstanceManager.java:150)
	at org.apache.catalina.core.StandardContext.listenerStart(StandardContext.java:4604)
	at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5150)
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1377)
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1367)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.lang.Thread.run(Thread.java:748)
2019-05-06 15:54:57,245 ERROR [Catalina-utility-1] [ContainerBase.[Standalone].[localhost].[/]] log Skipped installing application listeners due to previous error(s)

```

I followed the following instructions:

  https://apereo.atlassian.net/wiki/spaces/CASC/pages/103252576/Configuring+Confluence+with+JASIG+CAS+Client+for+Java+3.1

**Comment by hdeadman** [comment]
Are you sure you put the cas jars in the correct web-inf\lib folder for confluence?
`
/opt/atlassian/confluence/confluence/WEB-INF/lib $ ls ca*
cas-client-core-3.5.1.jar                   cas-client-integration-atlassian-3.5.1.jar
`

**Comment by hdeadman** [comment]
Also, the instructions are here: https://github.com/apereo/java-cas-client#atlassian-integration

**Comment by ukhangmu** [comment]
Hal,

Thanks for response.  I did read through the instructions you posted - thank you for that.  I thought I made some progress, but I am getting a different error now:

13-May-2019 16:35:14.431 WARNING [Catalina-utility-1] org.apache.catalina.loader.WebappClassLoaderBase.clearReferencesThreads The web application [ROOT] appears to have started a thread named [C3P0PooledConnectionPoolManager[identityToken->zzc70sa21dmzz5r1j2ifbj|84ccf56]-HelperThread-#1] but has failed to stop it. This is very likely to create a memory leak. Stack trace of thread:
 java.lang.Object.wait(Native Method)
 com.mchange.v2.async.ThreadPoolAsynchronousRunner$PoolThread.run(ThreadPoolAsynchronousRunner.java:683)
13-May-2019 16:35:14.432 WARNING [Catalina-utility-1] org.apache.catalina.loader.WebappClassLoaderBase.clearReferencesThreads The web application [ROOT] appears to have started a thread named [C3P0PooledConnectionPoolManager[identityToken->zzc70sa21dmzz5r1j2ifbj|84ccf56]-HelperThread-#2] but has failed to stop it. This is very likely to create a memory leak. Stack trace of thread:
 java.lang.Object.wait(Native Method)
 com.mchange.v2.async.ThreadPoolAsynchronousRunner$PoolThread.run(ThreadPoolAsynchronousRunner.java:683)
13-May-2019 16:35:14.433 WARNING [Catalina-utility-1] org.apache.catalina.loader.WebappClassLoaderBase.clearReferencesThreads The web application [ROOT] appears to have started a thread named [process-utils:idle] but has failed to stop it. This is very likely to create a memory leak. Stack trace of thread:
 sun.misc.Unsafe.park(Native Method)
 java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
 java.util.concurrent.SynchronousQueue$TransferStack.awaitFulfill(SynchronousQueue.java:458)
 java.util.concurrent.SynchronousQueue$TransferStack.transfer(SynchronousQueue.java:362)
 java.util.concurrent.SynchronousQueue.take(SynchronousQueue.java:924)
 java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1074)
 java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1134)
 java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
 java.lang.Thread.run(Thread.java:748)

Any suggestions?  Thanks
